libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -310,18 +310,13 @@ void CompilerUtils::abiEncode(
 {
 	// stack: <$value0> <$value1> ... <$value(n-1)> <$headStart>
 
-	vector<string> variables;
-	size_t numValues = sizeOnStack(_givenTypes);
-	for (size_t i = 0; i < numValues; ++i)
-			variables.push_back("$value" + to_string(i));
-	variables.push_back("$headStart");
-
-	ABIFunctions funs;
-	string routine = funs.tupleEncoder(_givenTypes, _targetTypes, _encodeAsLibraryTypes);
-	routine += funs.requestedFunctions();
-	m_context.appendInlineAssembly("{" + routine + "}", variables);
-	// Remove everyhing except for "value0" / the final memory pointer.
-	popStackSlots(numValues);
+	auto ret = m_context.pushNewTag();
+	moveIntoStack(sizeOnStack(_givenTypes) + 1);
+
+	string encoderName = m_context.abiFunctions().tupleEncoder(_givenTypes, _targetTypes, _encodeAsLibraryTypes);
+	m_context.appendJumpTo(m_context.namedTag(encoderName));
+	m_context.adjustStackOffset(-int(sizeOnStack(_givenTypes)) - 1);
+	m_context << ret.tag();
```

Comment:
`ret` doesn't seem to be passed to anything and such is just a stale tag?

Reply:
`pushNewTag` pushes a reference to it onto the stack and also returns it. `m_context << ret.tag()` appends the corresponding `jumpdest`.

Reply:
But what where is the jump?

Reply:
it is the return tag for a function call.

Reply:
This is super confusing tbh 😉 

Perhaps we could have a helper for this whole construct, `m_context.callFunction(string, stack_items)`.

Reply:
Yes it is confusing, and another reason why we should move to IULIA.
We have something similar called `callLowLevelFunction`.

Reply:
... but I think it cannot be used in this context.